tests/postgres_tests/test_array.py

```diff
@@ -173,12 +173,32 @@ def test_in(self):
             self.objs[:2]
         )
 
+    @unittest.expectedFailure
```

Comment:
These two tests test for another use case of this feature that I've been unable to implement.

Specifically, you can now provide a F object that refers to an array when doing filters on an array field, which is tested and working, however you cannot use an F object in constructing an array that goes into the filter.

I think it is better to demonstrate that this extension to the general feature doesn't work, but I'll defer to the core team's judgement on that.